### PR TITLE
DCS-1264 new endpoint for status and reasons data

### PR DESCRIPTION
**Pull request by GurnankCheema**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/resources/ImprisonmentReasonsResource.kt [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+        ]
+      ),
+    ]
+  )
+  @GetMapping(path = ["/imprisonment-statuses"], produces = [MediaType.APPLICATION_JSON_VALUE])
+  fun getStatuses(): List<ImprisonmentStatus> = imprisonmentStatuses
+}
+
+data class MovementReason(
+  val description: String? = "",
```
Comment (GurnankCheema) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/resources/ImprisonmentReasonsResource.kt:66]: assigning a default of "" overcomes the need to include 'description' in the objects that don't need it

Comment (bell-pa) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/resources/ImprisonmentReasonsResource.kt:66]: That should probably be `= null`, not `= ""`

Comment (bell-pa) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/resources/ImprisonmentReasonsResource.kt:66]: Add `@JsonInclude(JsonInclude.Include.NON_NULL)` to the MovementReason and ImprisonmentStatus classes.
Also, add some swagger annotations to the fields to document their purpose / meaning.

**Review by bell-pa — APPROVED** [review]
LGTM

**Pull request merged**